### PR TITLE
added documentation on common controller parameters

### PR DESCRIPTION
**Pull request by saikishor**
I've add some minor documentation update on the common controller parameters for every ROS2 Controller. 
@christophfroehlich FYI

### CODE REVIEW

File: doc/controllers_index.rst [hunk 1]
```diff
@@ -72,3 +72,12 @@ In the sense of ros2_control, broadcasters are still controllers using the same
    IMU Sensor Broadcaster <../imu_sensor_broadcaster/doc/userdoc.rst>
    Joint State Broadcaster <../joint_state_broadcaster/doc/userdoc.rst>
    Range Sensor Broadcaster <../range_sensor_broadcaster/doc/userdoc.rst>
+
+
+Common Controller Parameters
+****************************
+
+Every controller and broadcaster have a few common parameters that are necessary upon initialization. These parameters are `update_rate` and `is_async`.
```
Comment (christophfroehlich) [inline doc/controllers_index.rst:80]:
```suggestion
Every controller and broadcaster has a few common parameters that are optional upon initialization:
```

**Review by christophfroehlich — COMMENTED** [review]
Thanks. These parameters are optional, not necessary? or do you mean it is necessary that they are set before init if desired?

File: doc/controllers_index.rst (hunk 1 above)
Comment (saikishor) [reply doc/controllers_index.rst:80]: These parameters are needed to be set before configuring the controller.

Comment (christophfroehlich) [reply doc/controllers_index.rst:80]: I updated my suggestion.

I saw now that the docs lack of proper description of the lifecycle changes/transitions (nothing to link to).

File: doc/controllers_index.rst [hunk 2]
```diff
@@ -72,3 +72,12 @@ In the sense of ros2_control, broadcasters are still controllers using the same
...
+
+* `update_rate`: An unsigned integer parameter representing the rate at which every controller/broadcaster runs its update cycle. When unspecified, they run at the same frequency as the controller_manager.
+* `is_async`: A boolean parameter that is needed to specify if the controller update needs to run asynchronously.
```
Comment (christophfroehlich) [inline doc/controllers_index.rst:83]:
```suggestion
* ``update_rate``: An unsigned integer parameter representing the rate at which every controller/broadcaster runs its update cycle. When unspecified, they run at the same frequency as the controller_manager.
* ``is_async``: A boolean parameter that is needed to specify if the controller update needs to run asynchronously.
```

File: doc/controllers_index.rst (hunk 1 above)
Comment (saikishor) [reply doc/controllers_index.rst:80]: Hello, I reformatted a bit. Can you verify?

Every controller and broadcaster has a few common parameters. They are optional, but if needed they have to be set before ``onConfigure`` transition to ``inactive`` state, see `lifecycle documents <https://design.ros2.org/articles/node_lifecycle.html>`__. Once the controllers are already loaded, this transition is done using the ``configure_controller`` service of the controller_manager:

File: doc/controllers_index.rst [hunk 3]
```diff
@@ -72,3 +72,12 @@ In the sense of ros2_control, broadcasters are still controllers using the same
    IMU Sensor Broadcaster <../imu_sensor_broadcaster/doc/userdoc.rst>
    Joint State Broadcaster <../joint_state_broadcaster/doc/userdoc.rst>
    Range Sensor Broadcaster <../range_sensor_broadcaster/doc/userdoc.rst>
+
+
+Common Controller Parameters
+****************************
+
+Every controller and broadcaster has a few common parameters. They are optional, but if needed they have to be set before ``onConfigure`` transition to ``inactive`` state, see `lifecycle documents <https://design.ros2.org/articles/node_lifecycle.html>`__. Once the controllers are already loaded, this transition is done using the `configure_controller` service of the controller_manager.
```
Comment (christophfroehlich) [inline doc/controllers_index.rst:80]:
```suggestion
Every controller and broadcaster has a few common parameters. They are optional, but if needed they have to be set before ``onConfigure`` transition to ``inactive`` state, see `lifecycle documents <https://design.ros2.org/articles/node_lifecycle.html>`__. Once the controllers are already loaded, this transition is done using the service ``configure_controller`` of the controller_manager.
```

**Review by christophfroehlich — APPROVED** [review]
LGTM now.

**Review by bmagyar — APPROVED** [review]

**Pull request merged**